.travis.yml

```diff
@@ -25,3 +25,12 @@ matrix:
   allow_failures:
     - rvm: ruby-head
     - gemfile: gemfiles/rails_dev.gemfile
+before_install:
+  # https://docs.travis-ci.com/user/languages/ruby/#bundler
+  - RUBY_VERSION=$(ruby -v | grep -o -E '[0-9]\.[0-9]' | tr -d '.')
+  # Bundler for greater than Ruby 2.3 version
+  - if [ $RUBY_VERSION -gt 23 ]; then gem update --system; fi
+  - if [ $RUBY_VERSION -gt 23 ]; then gem install bundler -v '>= 2'; fi
+  # Bundler for less than Ruby 2.3 version or Rails 4.2
```

Comment:
In before_install step, we can't get Rails version. 
So, I think it might be good to use custom Jobs for ruby, rails, bundle dependencies.